cmd/pomerium-cli/cache_test.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+// creating Mock Cache operations
+type MockCache struct {
+	mock.Mock
+}
+
+// mock func for cache.ExecCredentialsPath()
+func (m *MockCache) ExecCredentialsPath() (string, error) {
+	args := m.Called()
+	return args.String(0), args.Error(1)
+}
+
+func TestClearAllCachedCredentials(t *testing.T) {
+	filePath := t.TempDir()
+	//mock object
+	mockCache := new(MockCache)
+
+	//set up dir expectation for ExecCredentialsPath
+	mockCache.On(" ExecCredentialsPath").Return(filePath+"/tmp/cache", nil)
+	_, err := os.Create(filePath + "/tmp/cache/file1.json")
+	if err != nil {
+		t.Log(err)
+		return
+	}
+	_, err = os.Create(filePath + "/tmp/cache/file2.json")
+	if err != nil {
+		t.Log(err)
+		return
+	}
+
+	err = clearAllCachedCredentials()
+	if err != nil {
+		t.Log(err)
+		return
+	}
+
+	// Verify that the cache directory is empty
+	dir, err := os.Open(filePath + "/tmp/cache")
+	assert.NoError(t, err)
+	defer dir.Close()
+
+	files, err := dir.Readdir(0)
+	assert.Error(t, err)
+	assert.Equal(t, 0, len(files))
+}
+
+func TestCachedCredentialPath(t *testing.T) {
+	//machine filepath name declared when running tests
+	pcPath := "/home/dell/.cache"
```

Comment:
This test has the same problem as above: the behavior under test depends on the return value of ExecCredentialsPath(). We would need to make sure that this test can pass when run by GitHub Actions and when run by a developer locally.

Reply:
thats true, the ci checks won't pass .